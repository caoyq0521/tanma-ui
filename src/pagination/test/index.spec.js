import { createLocalVue, shallowMount, mount } from '@vue/test-utils';
import TmPagination from '../pagination.vue';
import ElementUI, { Select } from 'element-ui';

const localVue = createLocalVue();
localVue.use(ElementUI);

describe('tmPagination', () => {
  it('create', () => {
    const wrapper = shallowMount(TmPagination, {
      localVue
    });

    expect(wrapper.find('.tm-pagination__total').exists()).toBe(true);
    expect(wrapper.find('.button-prev').exists()).toBe(true);
    expect(wrapper.find('.tm-pagination__pager').exists()).toBe(true);
    expect(wrapper.find('.button-next').exists()).toBe(true);
    expect(wrapper.find('.tm-pagination__sizes').exists()).toBe(true);
  });

  it('easy', () => {
    const wrapper = shallowMount(TmPagination, {
      propsData: {
        easy: true
      },
      localVue
    });

    expect(wrapper.find('.interval').exists()).toBe(true);
  });

  it('pageSize', () => {
    const wrapper = shallowMount(TmPagination, {
      propsData: {
        pageSize: 20,
        total: 100
      },
      localVue
    });

    expect(wrapper.findAll('span.number')).toHaveLength(5);
  });

  it('pagerCount', () => {
    const wrapper = shallowMount(TmPagination, {
      propsData: {
        pageSize: 20,
        total: 1000,
        pagerCount: 21
      },
      localVue
    });

    expect(wrapper.findAll('span.number')).toHaveLength(21);
  });

  it('currentPage', async () => {
    const wrapper = shallowMount(TmPagination, {
      propsData: {
        pageSize: 20,
        total: 200,
        currentPage: 3
      },
      localVue
    });

    expect(wrapper.find('span.number.active').text()).toBe('3');
  });

  it('currentPage: NaN', () => {
    const wrapper = shallowMount(TmPagination, {
     propsData: {
      pageSize: 20,
      total: 200,
      currentPage: NaN
     },
     localVue
    });

    expect(wrapper.find('span.number.active').text()).toBe('1');
    expect(wrapper.findAll('span.number')).toHaveLength(7);
  });

  it('buttonPrev and buttonNext: click', async () => {
    const spy = jest.spyOn(TmPagination.methods, "handlePrev");
    const wrapper = shallowMount(TmPagination, {
      propsData: {
        total: 200
      },
      localVue
    });

    expect(wrapper.vm.internalCurrentPage).toBe(1);

    await wrapper.find('.button-next').trigger('click');
    expect(wrapper.vm.internalCurrentPage).toBe(2);

    await wrapper.find('.button-prev').trigger('click');
    expect(wrapper.vm.internalCurrentPage).toBe(1);
    // console.log(spy);
    // expect(spy).toHaveBeenCalledWith(1);

  });

  it('pageSizes', async () => {
    const wrapper = mount(TmPagination, {
      propsData: {
        pageSizes: [10, 30, 35, 40],
        pageSize: 10
      }
    });

    // console.log(wrapper.vm.$el.querySelector('.tm-pagination__sizes'));

    // const bar = wrapper.findComponent(Select);
    // console.log("caoyq0521 ~ file: index.spec.js ~ line 113 ~ it ~ bar", bar)

    console.log(wrapper.find('.el-select'));

    await wrapper.find('.el-select').trigger('click');

    // console.log(wrapper.find('.tm-pagination__sizes-select-option'));
    // expect(wrapper.find('el-select-dropdown__item.selected>span').text()).toMatch('10');
  });
})
  