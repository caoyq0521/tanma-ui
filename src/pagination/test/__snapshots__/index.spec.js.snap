// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tmPagination renders correctly 1`] = `
<div class="tm-pagination"><span class="tm-pagination__total">共0条</span> <span disabled="disabled" class="button-prev"><i class="iconfont icon-zuojiantou"></i></span>
  <div class="tm-pagination__pager"><span class="number active">
        1
      </span>
    <!---->
    <!---->
    <!---->
  </div> <span disabled="disabled" class="button-next"><i class="iconfont icon-youjiantou"></i></span> <span class="tm-pagination__sizes"><el-select-stub value="20" autocomplete="off" disabled="true" multiplelimit="0" placeholder="请选择" valuekey="value" popperappendtobody="true"><el-option-stub value="20" label="20条/页"></el-option-stub><el-option-stub value="30" label="30条/页"></el-option-stub><el-option-stub value="40" label="40条/页"></el-option-stub><el-option-stub value="50" label="50条/页"></el-option-stub><el-option-stub value="100" label="100条/页"></el-option-stub></el-select-stub></span>
</div>
`;
